llm_inference.py

```diff
@@ -420,6 +429,109 @@ def reshape_model_outputs(outputs: List[str], input_batch_size: int) -> List[Lis
 
         return outputs
 
+class API_LLM(object):
+    """
+    Custom class for interacting with the LLMs via APIs.
+    """
+
+    def __init__(self, args: InferenceArguments):
+
+        self.args = args
+
+        from langchain.llms import Cohere, OpenAI
+        from langchain.chat_models import ChatOpenAI
+        
+        from api_secrets import COHERE_API_KEY, OPENAI_API_KEY
```

Comment:
Is there a particular reason for putting the inputs within the model init? Only advantage I could see is that some of the dependencies may not be necessary?

Reply:
Yeah, that was the only reason. I figured it 'encapuslated' the API model class a bit better.